### DISCUSSION TITLE
Drop integration plugins for Vite, Nuxt, Webpack and other bundlers in favor of  CLI.

### DISCUSSION THREAD

**Discussion by Machy8** [stylify/packages#274] [General]
Stylify can be integrated through `stylifyVite`, `stylifyNuxt`, `stylifyWebpack` and other plugins into random frameworks.

This means:
- One step implementation, but...
- This works only for simple cases
- More complexity
- More dependencies
- More things to study for installation
- More things to maintain
- Configuration different per bundler
- Random bugs because of the colision/race conditions during compliation https://github.com/stylify/packages/issues/266, https://github.com/stylify/packages/issues/255, https://github.com/stylify/packages/issues/251
- Random bugs because of how the configs are merged (for example Nuxt have 3 ways for configuring plugins => plugin have to handle all of them, otherwise it causes wtf moments).

## Why?
Because Stylify is a Compiler and can work as a Bundler on it's own, it doesn't require bundlers. 
Therefore, all these plugins could be deprecated and the initialization would be always the same:
1. Install `@stylify/cli`
2. Add command into package.json `"dev": "stylify dev & vite dev"`
3. Add path to generated CSS
4. (optional) Create additional config, that will always have the same layout
5. (optional) Prefixing can be solved through hooks, if necessary.

It would generate one or multiple CSS files that can be imported as needed.

## Bundlers are basically "obsolete" and the H in HMR stands for Headache
- First it was Webpack. Then Rollup and Phing(for php). Now Vite, Esbuild, Rustpack and the Turbopack is on it's way. 
- There is a lot of Bundlers
- The CLI would basically "remove" the strugle of "which plugin to use" and "how to configure it" and "why it doesn't work with my version of bundler" because, it would be completely separated.

Bundlers and are used for the compilation of assets (SCSS, Vue, JSX, Svelte, sometime images, other stuff) in JavaScript runtimes. This will probably (hopefully change):
- CSS can do a lot of things SCSS was used for
- Javascript may get some types in the future, Typescript will be slowly obsolete too
- Frameworks like Symfony are leaving bundlers
- Importmaps have a good support

**Comment by katriellucas** [comment]
I think this is a valid strategy, in [panda css](https://panda-css.com/docs/installation/cli) they use the CLI approach as well, they do have sections teaching how to install in any framework/bundler, but there is no specific plugins: 

![image](https://github.com/stylify/packages/assets/54755295/cb19ad5a-dfa4-4998-a021-9e0c725b1e5b)

**Reply by Machy8** [reply]
Thanks! I haven't heard about this one. I will go through their docs but it seems, it's very similar approach to what I would do.
